e2etest/bgptests/bgp.go

```diff
@@ -1376,6 +1378,94 @@ var _ = ginkgo.Describe("BGP", func() {
 		})
 	})
 
+	table.DescribeTable("A service of protocol load balancer should work with two protocols", func(pairingIPFamily ipfamily.Family, poolAddresses []string, tweak testservice.Tweak) {
+		var allNodes *corev1.NodeList
+		resources := metallbconfig.ClusterResources{
+			Pools: []metallbv1beta1.IPPool{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "bgp-test",
+					},
+					Spec: metallbv1beta1.IPPoolSpec{
+						Addresses: poolAddresses,
+					},
+				},
+			},
+			Peers:   metallb.PeersForContainers(FRRContainers, pairingIPFamily),
+			BGPAdvs: []metallbv1beta1.BGPAdvertisement{emptyBGPAdvertisement},
+		}
+
+		for _, c := range FRRContainers {
+			err := frrcontainer.PairWithNodes(cs, c, pairingIPFamily)
+			framework.ExpectNoError(err)
+		}
+
+		err := ConfigUpdater.Update(resources)
+		framework.ExpectNoError(err)
+
+		for _, c := range FRRContainers {
+			validateFRRPeeredWithNodes(cs, c, pairingIPFamily)
+		}
+
+		allNodes, err = cs.CoreV1().Nodes().List(context.Background(), metav1.ListOptions{})
+		framework.ExpectNoError(err)
+
+		svc, _ := testservice.CreateWithBackend(cs, f.Namespace.Name, "external-local-lb", tweak)
+		defer testservice.Delete(cs, svc)
+
+		validateDesiredLB(svc)
+
+		for _, c := range FRRContainers {
+			validateService(cs, svc, allNodes.Items, c)
+		}
+
+		checkServiceL2 := func() error {
+			for _, ip := range svc.Status.LoadBalancer.Ingress {
+				ingressIP := e2eservice.GetIngressPoint(&ip)
+				err := mac.RequestAddressResolution(ingressIP, executor.Host)
+				if err != nil {
+					return err
+				}
+			}
+			return nil
+		}
+
+		ginkgo.By("Checking the service is not reacheable via L2")
+		Consistently(checkServiceL2, 3*time.Second, 1*time.Second).Should(Not(BeNil()))
+
+		ginkgo.By("Creating the l2 advertisement")
+		l2Advertisement := metallbv1beta1.L2Advertisement{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "l2adv",
+				Namespace: metallb.Namespace,
+			},
+		}
+
+		err = ConfigUpdater.Client().Create(context.Background(), &l2Advertisement)
+		framework.ExpectNoError(err)
+
+		Eventually(func() error {
+			return testservice.ValidateL2(svc)
```

Comment:
Does it make sense to re-run the function "validateService" to assert that
creating the l2Adv didn't break anything?

Reply:
not sure, but it's cheap enough (in terms of test running time), and it won't hurt

Reply:
done